Sming/SmingCore/DateTime.cpp

```diff
@@ -379,7 +379,7 @@ String DateTime::format(String sFormat)
 void DateTime::calcDayOfYear()
 {
 	DayofYear = 0;
-	for(unsigned int i = 0; i < Month; ++i) {
+	for(int i = 0; i < Month; ++i) {
```

Comment:
Just for my understanding: Why not use `unsigned` here?

Reply:
`Month` is signed (`int8_t`) as are all the member variables and methods. I can re-write the `DateTime` class using unsigned values, which is rather more extensive and would be a separate PR.
This seemed like an easier solution unless there's actually a problem with `DateTime`. What'cha wanna do?

Reply:
> This seemed like an easier solution unless there's actually a problem with `DateTime`

Then it is ok.

Reply:
> unless there's actually a problem with `DateTime`

Had a read through and found a few things. Will submit separate PR for this.